src/intel/sse.rs

```diff
@@ -0,0 +1,1400 @@
+#![cfg(target_feature = "sse")]
+
+use super::*;
+
+// TODO: load and store m128i values. The operations are technically part of
+// `sse`, but all actual usage of `m128i` is from `sse2`, so we won't tackle
+// that right away.
+
+/// Lanewise `a + b`.
+/// ```
+/// # use safe_arch::*;
+/// let a = m128::from_array([1.0, 2.0, 3.0, 4.0]);
+/// let b = m128::from_array([5.0, 6.0, 7.0, 8.5]);
+/// let c = add_m128(a, b).to_array();
+/// assert_eq!(c, [6.0, 8.0, 10.0, 12.5]);
+/// ```
+#[must_use]
+#[inline(always)]
+pub fn add_m128(a: m128, b: m128) -> m128 {
+  m128(unsafe { _mm_add_ps(a.0, b.0) })
+}
+
+/// Low lane `a + b`, other lanes unchanged.
+/// ```
+/// # use safe_arch::*;
+/// let a = m128::from_array([1.0, 2.0, 3.0, 4.0]);
+/// let b = m128::from_array([5.0, 6.0, 7.0, 8.5]);
+/// let c = add_m128_low(a, b).to_array();
+/// assert_eq!(c, [6.0, 2.0, 3.0, 4.0]);
+/// ```
+#[must_use]
+#[inline(always)]
+pub fn add_m128_low(a: m128, b: m128) -> m128 {
```

Comment:
it's because only the lowest lane is changed.

basically i need a name for "low lane only" and also it has to be a very small number of characters.

Reply:
You could go for `_v` and `_s` for vector and scalar. `low` is… weird. It can be understood as »lowest lane«, but common terminology for registers is that »low« means the entire lower half, i.e. two lanes here. Like `AX` vs. `AL`, `movehl_ps`, etc.

Reply:
Yeah, _s is better actually. and shorter!